switcher.js

```diff
@@ -201,33 +228,141 @@ class Switcher {
+              return prev;
+            }
+
+            // This regex will match [.*] and extract interior characters as the second array item
+            // matches[0] = [AAAA]
+            // matches[1] = AAAA
+            const matches = curr.match(/\[(.*)\]/);
+
+            // If this is a profile name, push a new profile object to the array with the name
+            if (matches) {
+              // The config file user the pattern [profile profileName], need to strip the word profile
+              const [, name] = matches.map(match => match.replace(/profile/i, '').trim());
+              const existingProfile = prev.find(p =>
+                p.name !== undefined &&
+                Object.keys(p).length < 5
+              );
+              if (!existingProfile) {
+                prev.push({ name });
+              }
+              return prev;
+            }
+
+            /*
+            * At this point this line is neither blank nor a profile name
+            * Which means it's either an AWS profile ID or a configuration which are are formatted like:
+            *  output = OUTPUT
+            *  region = REGION
+            * With spaces in between the '=' and the key/value
+            */
+
+            // Split on '=' and then remove any spaces
+            // [0] === 'output'  || [0] === 'region'
+            // [1] === OUTPUT    || [1] === REGION
+            const [key, value] = curr.split('=').map(section => section.replace(' ', ''));
+
+            // Find an existing array object with a defined name and less then 5 object properties
+            // Each profile object should end up with 5 properties:
+            // [name, aws_access_key_id, aws_secret_access_key, region, output]
+            // Only the most recent profile object will have a defined name and less then 5 properties
+            const profile = prev.find(p =>
+              p.name !== undefined &&
+              Object.keys(p).length < 5
+            );
+
+            profile[key] = value;
+            return prev;
+          }, profileData || []); // Pass in the existing profileDate of an empty array to start the reduce
+          return resolve(data);
+        });
+      });
+    }
+    return Promise.resolve(profileData);
+  }
+
   /**
    * Writes the credentials data back to the credentials file
    *
    * @param {Object[]} profile - Array of profile objects
    * @param {String} profile.name                   - The profile's name (no square brackets)
    * @param {String} profile.aws_access_key_id      - The profile's access key ID (from AWS)
    * @param {String} profile.aws_secret_access_key  - The profile's secret access key (from AWS)
+   * @param {String} profile.output  - The profile's output type
+   * @param {String} profile.region  - The profile's region location
    * @returns {Promise.<undefined>} - Returns an empty resolved Promise
    * @reject {Error} - Error writing the credentials file
    */
   _writeBackCredentialsData(data) {
```

Comment:
I'd rather that there was something like a `_writeFiles` function that was called, which in turn called both `_writeBackCredentialsData` and a `_writeBackConfigFileData` function and handled any errors, that way the config file logic and the credentials file logic stay separate and clean.